modules/schematics/src/create-action-migration/index.ts

```diff
@@ -0,0 +1,206 @@
+              statement.declarationList.declarations.filter(
+                node => (node.name as ts.Identifier).text === typeProperty
+              ).length
+          );
+
+          const typeValue = typeVariable
+            .map(statement => statement.declarationList.declarations)
+            .map(decl => (decl[0].initializer as ts.Identifier).text)[0];
+
+          if (!typeValue) {
+            return undefined;
+          }
+
+          const payloadType = getConstructorPayloadType(
+            host,
+            path,
+            actionClass
+          );
+          if (payloadType) {
+            importProps = true;
+          }
+
+          const createActionDefinition = composeCreateAction(
+            typeValue,
+            payloadType,
+            actionClass
+          );
+
+          if (createActionDefinition.includes('props')) {
+            importProps = true;
+          }
+
+          return [
+            new RemoveChange(
+              sourceFile.fileName,
+              typeVariable[0].getStart(sourceFile),
+              typeVariable[0].getEnd()
+            ),
+            new RemoveChange(
+              sourceFile.fileName,
+              actionClass.getStart(sourceFile),
+              actionClass.getEnd()
+            ),
+            new InsertChange(
+              path,
+              typeVariable[0].getStart(sourceFile),
+              createActionDefinition
+            ),
+          ];
+        })
+        .filter(Boolean) as []).reduce(
+        (imports, curr) => imports.concat(curr),
+        []
+      );
+
+      let importChanges: (ReplaceChange | RemoveChange | Change)[] = [];
+      const imports = [];
+      if (changes.length > 0) {
+        imports.push('createAction');
+        if (importProps) {
+          imports.push('props');
+        }
+        importChanges = replaceImport(
+          sourceFile,
+          path,
+          '@ngrx/store',
+          'Action',
+          imports.join(', ')
+        );
+      }
+
+      return commitChanges(host, sourceFile.fileName, [
+        ...changes,
+        ...importChanges,
+      ]);
+    });
+}
+
+const isActionImported = (sourceFile: ts.SourceFile): boolean => {
+  return !!sourceFile.statements
+    .filter(ts.isImportDeclaration)
+    .filter(({ moduleSpecifier }) =>
+      moduleSpecifier.getText(sourceFile).includes('@ngrx/store')
+    )
+    .filter(p => {
+      return (p.importClause!
+        .namedBindings! as ts.NamedImports).elements.filter(spec => {
+        if (spec.name.text) {
+          return spec.name.text === 'Action';
+        }
+      }).length;
+    }).length;
+};
+
+const getActionClasses = (sourceFile: ts.SourceFile): ts.ClassDeclaration[] => {
+  return sourceFile.statements
+    .filter(ts.isClassDeclaration)
+    .filter(
+      statement => statement.heritageClauses && statement.heritageClauses.length
+    )
+    .filter(statement => {
+      return statement.heritageClauses!.filter(clause => {
```

Comment:
I think it would be easier to read if we would use `some` in favor of `filter([].length)`

Reply:
Hey Tim, do you mean something like this?
```javascript
      return statement.heritageClauses!.some(clause => {
        return (clause.types || []).some(type => {
          return (
            type.expression && type.expression.getText(sourceFile) === 'Action'
          );
        });
      });
```